### ISSUE TITLE
window not defined in content shared file in v0.19

### ISSUE THREAD

**Issue by mehimanshupatil** [wxt-dev/wxt#864]
### Describe the bug

I have a content script which has a react code.
I have created constant file from which i export window variable.
the `wxt prepare` command fails 

<details>
  <summary>Error logs</summary>

```
 > wxt prepare


WXT 0.19.0                                                                                         5:51:11 PM
ℹ Generating types...                                                                             5:51:13 PM
✖ Command failed after 3.837 s                                                                    5:51:15 PM

 ERROR  window is not defined                                                                      5:51:15 PM

  at entrypoints/content/constant.ts:1:21
  at ViteNodeRunner.runModule (node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.12/node_modules/vite-node/dist/client.mjs:399:11)
  at ViteNodeRunner.directRequest (node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.12/node_modules/vite-node/dist/client.mjs:381:16)
  at ViteNodeRunner.cachedRequest (node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.12/node_modules/vite-node/dist/client.mjs:206:14)
  at ViteNodeRunner.dependencyRequest (node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.12/node_modules/vite-node/dist/client.mjs:259:12)
  at entrypoints/content/App.tsx:3:31
  at ViteNodeRunner.runModule (node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.12/node_modules/vite-node/dist/client.mjs:399:5)
  at ViteNodeRunner.directRequest (node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.12/node_modules/vite-node/dist/client.mjs:381:5)
  at ViteNodeRunner.cachedRequest (node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.12/node_modules/vite-node/dist/client.mjs:206:14)
  at ViteNodeRunner.dependencyRequest (node_modules/.pnpm/vite-node@2.0.4_@types+node@20.14.12/node_modules/vite-node/dist/client.mjs:259:12)
```
</details>

### Reproduction

https://github.com/mehimanshupatil/wxt-example

### Steps to reproduce

pnpm i

### System Info

```shell
System:
    OS: macOS 14.5
    CPU: (8) arm64 Apple M1
    Memory: 65.75 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
Binaries:
    Node: 20.15.0 - ~/.nvm/versions/node/v20.15.0/bin/node
    npm: 10.8.2 - ~/.nvm/versions/node/v20.15.0/bin/npm
    pnpm: 9.6.0 - ~/.nvm/versions/node/v20.15.0/bin/pnpm
    Watchman: 2024.06.10.00 - /opt/homebrew/bin/watchman
Browsers:
    Brave Browser: 127.1.68.128
    Edge: 127.0.2651.74
    Safari: 17.5
npmPackages:
    wxt: ^0.19.0 => 0.19.0
```

### Used Package Manager

pnpm

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/wxt-dev/wxt/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://wxt.dev/guide/installation.html).
- [X] Check that there isn't [already an issue](https://github.com/wxt-dev/wxt/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/wxt-dev/wxt/discussions) or join our [Discord Chat Server](https://discord.gg/ZFsZqGery9).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by aiktb** [comment]
I have the same issue, the side effect of the import statement is triggered in v0.19.0, but not in v.0.18.0.

**Comment by Timeraa** [comment]
Pinged @aklinker1 about this in the Discord but seems to be an issue from the move to tsup, same behaviour with `self`

**Comment by aklinker1** [comment]
@aiktb @Timeraa @mehimanshupatil Can you guys try out `wxt@0.19.1-alpha1` and let me know if it works for you?

Just an FYI - you shouldn't be using browser environment globals, like `window` or `self`, outside the `main` function of an entrypoint. In this new version, I made it so most browser environment globals are at least defined, but you will still run into issues with things like `window.location.href`, because I didn't implement all the APIs.

**Comment by aiktb** [comment]
> @aiktb @Timeraa @mehimanshupatil Can you guys try out `wxt@0.19.1-alpha1` and let me know if it works for you?
> 
> Just an FYI - you shouldn't be using browser environment globals, like `window` or `self`, outside the `main` function of an entrypoint. In this new version, I made it so most browser environment globals are at least defined, but you will still run into issues with things like `window.location.href`, because I didn't implement all the APIs.

Fixed.👍

Thank you for your answer, I have a Selector Class, in which the static member new a Class with a `document.querySelector` operation, and that is the problem.

**Comment by mehimanshupatil** [comment]
getting error when running prepare

```

WXT 0.19.1-alpha1                                                                           2:37:21 PM
ℹ Generating types...                                                                      2:37:21 PM
SETUP
TEARDOWN
✖ Command failed after 1.177 s                                                             2:37:22 PM

[2:37:22 PM]  ERROR  Only URLs with a scheme in: file, data, and node are supported by the default ESM loader. Received protocol 'virtual:'

  at throwIfUnsupportedURLScheme (node:internal/modules/esm/load:239:11)
  at defaultLoad (node:internal/modules/esm/load:130:3)
  at ModuleLoader.load (node:internal/modules/esm/loader:396:13)
  at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:278:56)
  at new ModuleJob (node:internal/modules/esm/module_job:67:26)
  at ModuleLoader.#createModuleJob (node:internal/modules/esm/loader:290:17)
  at ModuleLoader.getJobFromResolveResult (node:internal/modules/esm/loader:247:34)
  at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:228:17)
  at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:87:21)


```

**Comment by aklinker1** [comment]
@mehimanshupatil Interesting, could you share your project?

**Comment by mehimanshupatil** [comment]
@aklinker1 
its reproducible on Reproduction repo
https://github.com/mehimanshupatil/wxt-example

**Comment by aklinker1** [comment]
Yup, got it now, thanks. Looks like `vite-node` doesn't like the `wxt/client` import... For now, `wxt@0.19.1-alpha2` fixes this for your case, but I need to do additional work before releasing 0.19.1 based on this feedback and this thread: https://github.com/wxt-dev/wxt/discussions/866

**Comment by mehimanshupatil** [comment]
cool window error is resolved
another error related to scss file import
when scss file is imported in `index.tsx` error is thrown
`[sass] Cannot read properties of undefined (reading 'pop')`
have updated Reproduction repo to reproduce sass erorr

**Comment by aklinker1** [comment]
Ok, sounds good! Thanks for helping me test this out 🙏

**Comment by aklinker1** [comment]
@mehimanshupatil Sass error fixed in `0.19.1-alpha3`, confirmed with your latest reproduction.

**Comment by aklinker1** [comment]
Fixes released in [`v0.19.1`](https://github.com/wxt-dev/wxt/releases/tag/wxt-v0.19.1)
